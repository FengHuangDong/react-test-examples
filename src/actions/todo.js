import axios from 'axios';
import { REMOVE_TODO, ADD_TODO, SET_VISIBILITY_FILTER,FETCH_TODOS,ERR_FETCH_TODOS, TOGGLE_TODO } from "./actionsTypes";

export const addToList = (text) => {
  return {
    type: ADD_TODO,
    payload: text
  };
};

export const setVisibilityFilter = (type) => {
  return {
    type:SET_VISIBILITY_FILTER,
    payload:type
  }
}
export const toogleTodo = (id) => {
  return {
    type:TOGGLE_TODO,
    payload:id
  }
}
export const removeFromList = (text) => {
  return {
      type:REMOVE_TODO,
      payload: text
  }
};


export const fetchTodos = () => {
  return async (dispatch) => {
    try {
      let todos = await axios.get(`https://jsonplaceholder.typicode.com/todos`)
      dispatch({
        type: FETCH_TODOS,
        payload: todos
      })
    } catch (err) {
      debugger
      dispatch({
        type: ERR_FETCH_TODOS,
        payload: err
      })
    }
  }
}
