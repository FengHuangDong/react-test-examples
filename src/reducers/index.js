import todos from './todo';
import { combineReducers } from 'redux';
import { reducer as formReducer } from 'redux-form';
import visibilityFilter from './FilterReducer'

const rootReducer = combineReducers({
    todos,
    visibilityFilter,
    form: formReducer
});
export default rootReducer;
