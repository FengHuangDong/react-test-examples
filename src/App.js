import React, { Component } from 'react';
import './App.css';

import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import {addToList, fetchTodos} from './actions/todo';

import TodoInput from './components/TodoInput'
import TodoList from './components/TodoList'

export class App extends Component {
  componentDidMount(){
    this.props.fetchTodos()
  }
  mySubmit(values) {
    this.props.addToList(values.title)
  }

  render() {
    return (
      <div className="APP">
        <div style={{ marginTop: "80px"}} className="col-lg-10 offset-lg-1 col-md-10 col-sm-12 col-xs-12">
          <TodoInput onSubmit={this.mySubmit.bind(this)}/>
          <TodoList/>
        </div>
      </div>
    );
  }
}

export const mapDispatchToProps = dispatch => bindActionCreators({ addToList,fetchTodos }, dispatch)

export default connect(null, mapDispatchToProps)(App);
