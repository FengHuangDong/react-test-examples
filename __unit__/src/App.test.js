import React from 'react';
import configureMockStore from 'redux-mock-store';
import thunkMiddleware from 'redux-thunk';
import {App,mapDispatchToProps} from "../../src/App"
import {shallow} from 'enzyme'
import toJson from "enzyme-to-json"
import * as actionTypes from '../../src/actions/actionsTypes'

describe('mapDispatchToProps', () => {
  it('should action to be call when dispatch',async () => {
    const configMockStore = configureMockStore([thunkMiddleware]);
    const mockStore = configMockStore({});
    const dispatchMappedProps = mapDispatchToProps(mockStore.dispatch)
    await dispatchMappedProps.fetchTodos()
    expect(mockStore.getActions().map(action => action.type)).toEqual([actionTypes.FETCH_TODOS])
  })
});

describe('App Components', () => {
  let mockProps
  beforeEach(() => {
    mockProps = {
      fetchTodos:jest.fn(),
      addToList:jest.fn()
    }
  })
  it('should render the component', () => {
    const wrapper = shallow(<App {...mockProps}/>)
    expect(mockProps.fetchTodos).toHaveBeenCalled();
    expect(toJson(wrapper)).toMatchSnapshot();
  })
  it("should call addToList handle",() => {
    const wrapper = shallow(<App {...mockProps}/>)
    const instance = wrapper.instance()
    const mockArgs ={title:"test"}
    instance.mySubmit(mockArgs)
    expect(mockProps.addToList).toHaveBeenCalled();
  })
})
