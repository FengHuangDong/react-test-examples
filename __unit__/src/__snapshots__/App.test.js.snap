// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Components should render the component 1`] = `
<div
  className="APP"
>
  <div
    className="col-lg-10 offset-lg-1 col-md-10 col-sm-12 col-xs-12"
    style={
      Object {
        "marginTop": "80px",
      }
    }
  >
    <ReduxForm
      destroyOnUnmount={true}
      enableReinitialize={false}
      forceUnregisterOnUnmount={false}
      form="todo"
      getFormState={[Function]}
      keepDirtyOnReinitialize={false}
      onSubmit={[Function]}
      persistentSubmitErrors={false}
      pure={true}
      shouldAsyncValidate={[Function]}
      shouldError={[Function]}
      shouldValidate={[Function]}
      shouldWarn={[Function]}
      submitAsSideEffect={false}
      touchOnBlur={true}
      touchOnChange={false}
      updateUnregisteredFields={false}
    />
    <ConnectFunction />
  </div>
</div>
`;
